pkg/middleware/stored_session_test.go

```diff
@@ -17,9 +18,74 @@ import (
 	. "github.com/onsi/gomega"
 )
 
+type testLock struct {
+	locked          bool
+	obtainOnAttempt int
+	obtainAttempts  int
+	obtainError     error
+}
+
+func (l *testLock) Obtain(_ context.Context, _ time.Duration) error {
+	l.obtainAttempts++
+	if l.obtainAttempts < l.obtainOnAttempt {
+		return sessionsapi.ErrLockNotObtained
+	}
+	if l.obtainError != nil {
+		return l.obtainError
+	}
+	l.locked = true
+	return nil
+}
+
+func (l *testLock) Peek(_ context.Context) (bool, error) {
```

Comment:
With your new implementation, we can remove the need for a `Peek` function on the interface, right?